### ISSUE TITLE
Fix React Component Data Table -508 issue 

### ISSUE THREAD

**Issue by JanellC** [US-EPA-CAMD/PI-1#1513]
## Required Fix 

- [x] RF 3
- [x] RF 7 
- [x] RF 8

## Context 

See ticket Fix Data Table - 508 issue (TEST ONLY) #1494 for comments from 508 SME 

See [EPA Monitoring Plans Accessibility Report Additional.docx](https://github.com/US-EPA-CAMD/easey-ui/files/6632145/EPA.Monitoring.Plans.Accessibility.Report.Additional.docx) for how to fix and details 

RF 3: The sortable table does not have a programatic indicator to tell a screen reader when a column is sorted in ascending or descending order.

RF-7: The data tables in the application lack a programatic name:

RF-8: There are multiple tags with the same id when data tables are used on a page. For example:
6RmV1OFmK3-undefined
div id="cell-6RmV1OFmK3-undefined" role="gridcell" data-tag="allowRowEvents" class="sc-hKFxyN sc-eCApnc sc-jSFjdj iCxiXr domoiu dUbULe rdt_TableCell">
5
div id="cell-6RmV1OFmK3-undefined" role="gridcell" data-tag="allowRowEvents" class="sc-hKFxyN sc-eCApnc sc-jSFjdj iCxiXr domoiu dUbULe rdt_TableCell">

**Comment by JorjaComer** [comment]
@aprematta this ready for testing

**Comment by aprematta** [comment]
RF-7 is a pass, the table now has a programatic name through the aria label
RF-8 is a pass, looks like all id's are now unique

**Comment by aprematta** [comment]
RF-3 - ANDI and the screen readers are not picking up the sort. I think it's because the aria sort is not in the correct div. It's currently in the parent div to the column header:
<div class="sc-hKFxyN sc-eCApnc sc-iqAclL dSYVyP domoiu ikSTQe rdt_TableCol epa-active-element" aria-sort="ascending">
It should be in the same div as the column header
<div id="column-col1" role="columnheader" tabindex="0" class="sc-crzoAE cJjDWM rdt_TableCol_Sortable"><div class="epa-active-element">ORIS</div>

**Comment by aprematta** [comment]
Data cells are still not associated with the column headers. From what I can see, it's because the role=table is used with role=gridcell. all the cells should be modified to use role=cell instead.

"<div id="cell-l7AHSmCDlX-5" role="gridcell" data-tag="allowRowEvents" class="sc-hKFxyN sc-eCApnc sc-jSFjdj iCxiXr domoiu dUbULe rdt_TableCell"><div data-tag="allowRowEvents">Chickasaw</div></div>"

This issue is now fixed

**Comment by aprematta** [comment]
1 issue remains.....there is still no programmatic way for screen readers to tell if the columnheader is sorted in ascending or descending order

**Comment by aprematta** [comment]
This is not working on dev and appears to have reverted a bit. Issues are:
1. Sort (ascending /descending) is not communicated to the screen reader (aria sort is on the wrong div)
2. Cells are not associated to header cells (cells have the role of gridcell instead of cell)

Note that if comparison is needed, the table on CAMPD did pass the 508 test

**Comment by aprematta** [comment]
Cells are now associated to the header cells. However the Sort is still not communicated to the screenreader beccause the aria is not in the same div where the role of columnheader is. Note that if comparison is needed, the table on CAMPD did pass the 508 test.

**Comment by aprematta** [comment]
The table passes testing
